packages/bpk-component-modal/src/BpkModalV2/BpkModal.tsx

```diff
@@ -116,22 +116,29 @@ export const BpkModalV2 = (props: Props) => {
       ref.current?.showModal?.();
 
       const dialog = document.getElementById(`${id}`);
-      if (dialog) {
-        dialog.addEventListener('click', (event: Event) => {
-          const { target } = event;
-
-          if (target instanceof HTMLElement && target.id === `${id}`) {
-            ref.current?.close?.();
-          }
-        });
-      }
+      const dialogWithPolyfill = document.getElementById(`${id}-polyfill`);
```

Comment:
is the element with `id=${id}-polyfill` not just an extra element that will be rendered along with the `dialog`?
Looking at the code it doesn't seem like it's one or the other so wondering how `modal === dialog ? ref.current?.close?.() : onClose();` this works?

Reply:
not sure if there's a way to "hack" the dialog examples so that we can add it in Storybook 🤔

Reply:
or maybe rephrasing the question - if `dialog` HTML element is not supported, should we still be rendering it? 🤔

Reply:
It doesn't wrap it though - https://github.com/Skyscanner/backpack/blob/main/packages/bpk-component-modal/src/BpkModalV2/BpkModal.tsx#L156

Reply:
Sorry, I did not explain myself very well on this one. The wrapper is there on both supported/not supported. But for the unsupported it add the `id=${id}-polyfill`:  [https://github.com/Skyscanner/backpack/blob/main/packages/bpk-component-modal/src/BpkModalV2/BpkModal.tsx#L153](https://github.com/Skyscanner/backpack/blob/main/packages/bpk-component-modal/src/BpkModalV2/BpkModal.tsx#L153)

Reply:
Sorry, I got confused in my own code here. I updated my comment to reflect it.